components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml

```diff
@@ -267,84 +299,61 @@
 {
   var widgets = PhenoTips.widgets = PhenoTips.widgets || {};
   /* hack; otherwise loading order for push is wrong */
-  var consentsModule = {
-    createNew: function ()
-    {
-    }
-  };
+  var consentsModule = {};
   require(['jquery'], function ($)
   {
     consentsModule.createNew = function ()
     {
       return {
         consentIdSuffix: "-consent-checkbox",
+
         classVariables: {
           consentsBlock: null,
-          errorFunction: null,
-          successfulLoadCallback: null,
-          baseUrl: null,
+          errorBlock: null,
+          displayDetails: true,
+          consentChangeCallback: null,
           action: null
         },
 
         /**
          * @param callback function fired when the consent are successfully loaded and displayed
+         * @param errorBlock (optional) an HTML block which may have elements with the following IDs, which
+         *        (if present) are displayed and hidden depending on the state of consent data:
+         *        "#patient-consent-update" - supposedly an "update" button, hidden when any errors in consent data are present
+         *        ".view-title-required-not-granted" - a block shown when a required consent is not granted (in view mode)
+         *        ".view-title-none-granted" - a block shown when no consents have been granted (in view mode)
+         *        ".error_no_consent_data" - a block (supposedly with an error message) displayed when no consent data was passed to the script
```

Comment:
Because we are too lazy to annotate all? :) In this case, the function depends on external HTML block having certain elements with certain IDs/class names, which is not obvious for the user. The right way is to completely rewrite this, but in the rush I thought it is better to document and move on to refactoring more problematic areas